### DISCUSSION TITLE
How the heck does reflex update state when it is used in a component's f string?

### DISCUSSION THREAD

**Discussion by guidocalvano** [reflex-dev/reflex#2120] [Q&A] [answered]
It isn't really a bug or anything, but I am just so curious how you made that happen, and I figured, I can always ask... 

How the heck does reflex update state when it is used in a component's f string?

Am I correct in the following assertions?: Whenever you use a member of a rx.State it calls a function that stores its use, while it returns a template variable denoting string. The next component that is created is now associated with that State's member. Whenever the state is updated the template string is replaced with the State's member value.

This is the best I could come up with. So far your library seems awesome by the way.

**Comment by picklelo** [comment, marked as answer]
When using f-strings we override the `__format__` method within the `Var` class: https://github.com/reflex-dev/reflex/blob/6e71393ed5f29234071da2b43192fc12d9bdbbd7/reflex/vars.py#L278

So basically if you do something like `rx.text(f"{State.text} formatted")` it will render out to 
```js
<Text>
  {`${state.text} formatted`}
</Text>
```

So basically the frontend references the state variable, so whenever it gets updated the template string will update too.

**Comment by timmypidashev** [comment]
Pretty kewl, can't argue with that.
